spec.html

```diff
@@ -37526,120 +37526,121 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
             )
           </h1>
           <dl class="header">
-            <dt>description</dt>
-            <dd>It also has access to the _comparefn_ argument passed to the current invocation of the `sort` method.</dd>
           </dl>
           <emu-alg>
-            1. If _x_ and _y_ are both *undefined*, return *+0*<sub>𝔽</sub>.
-            1. If _x_ is *undefined*, return *1*<sub>𝔽</sub>.
-            1. If _y_ is *undefined*, return *-1*<sub>𝔽</sub>.
-            1. If _comparefn_ is not *undefined*, then
-              1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
-              1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
-              1. Return _v_.
-            1. [id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).
-            1. [id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).
-            1. Let _xSmaller_ be <emu-meta suppress-effects="user-code">IsLessThan(_xString_, _yString_, *true*)</emu-meta>.
-            1. If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.
-            1. Let _ySmaller_ be <emu-meta suppress-effects="user-code">IsLessThan(_yString_, _xString_, *true*)</emu-meta>.
-            1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
-            1. Return *+0*<sub>𝔽</sub>.
+            1. Let _items_ be a new empty List.
+            1. Let _k_ be 0.
+            1. Repeat, while _k_ &lt; _len_,
+              1. Let _Pk_ be ! ToString(𝔽(_k_)).
+              1. Let _kPresent_ be ? HasProperty(_obj_, _Pk_).
+              1. If _kPresent_ is *true*, then
+                1. Let _kValue_ be ? Get(_obj_, _Pk_).
+                1. Append _kValue_ to _items_.
+              1. Set _k_ to _k_ + 1.
+            1. Let _itemCount_ be the number of elements in _items_.
+            1. [id="step-array-sort"] Sort _items_ using an implementation-defined sequence of calls to _SortCompare_. If any such call returns an abrupt completion, stop before performing any further calls to _SortCompare_ or steps in this algorithm and return that completion.
+            1. Let _j_ be 0.
+            1. Repeat, while _j_ &lt; _itemCount_,
+              1. Perform ? Set(_obj_, ! ToString(𝔽(_j_)), _items_[_j_], *true*).
+              1. Set _j_ to _j_ + 1.
+            1. Repeat, while _j_ &lt; _len_,
+              1. Perform ? DeletePropertyOrThrow(_obj_, ! ToString(𝔽(_j_))).
+              1. Set _j_ to _j_ + 1.
+            1. Return _obj_.
           </emu-alg>
+          <p>The <dfn id="sort-order">sort order</dfn> is the ordering of _items_ after completion of step <emu-xref href="#step-array-sort"></emu-xref> of the algorithm above. If _SortCompare_ is not a consistent comparator for the elements of _items_ (see below), the sort order is implementation-defined. The algorithm that invokes SortIndexedProperties may specify broader conditions under which the sort order is implementation-defined.</p>
```

Comment:
Nit: this splits up the description of the sort order in a weird way. Previously it was ordered as "the sort order is implementation-defined in these cases; otherwise the sort order must satisfy the following properties:". Now it's "the sort order is implementation-defined in these cases", then a new subclause with a whole algorithm in it, then "if it's not implementation-defined the sort order must satisfy the following properties:".

I imagine this is because the two sections of constraints on sort order refer to specific steps of different algorithms, and so the constraints are grouped with the corresponding algorithms. Even so, it seems like they should be contiguous, as it was previously. I think the way I'd do it would be to place this section ("it must satisfy all of the following conditions") immediately following the "implementation-defined if any of the following conditions is true" section, and replace "after completion of step X of the algorithm above" with "after completion of step X of SortIndexedProperties". (And also move the definition of "sort order" to just prior to the first use of it.)

Reply:
> Nit: this splits up the description of the sort order in a weird way.

Yes, that seemed necessary to address your objection.

> I think the way I'd do it would be to place this section ("it must satisfy all of the following conditions") immediately following the "implementation-defined if any of the following conditions is true" section, [...] (And also move the definition of "sort order" to just prior to the first use of it.)

The 'sort order' definition and the "it must satisfy all of the following conditions" chunk specify the required behavior of SortIndexedProperties, in terms of aliases that only occur in SortIndexedProperties. You're suggesting that that specification be moved out of the SortIndexedProperties clause?

Reply:
> You're suggesting that that specification be moved out of the SortIndexedProperties clause?

Yup. I would also be OK with moving the other half into the SortIndexedProperties clause, I suppose, though that seems marginally worse.

Note that the "The sort order is implementation-defined if any of the following conditions is true" part is _also_ specifying the behavior of SortIndexedProperties. As I said, I recognize that the "implementation-defined if" part is in terms of aliases in a different algorithm than the "it must satisfy all of the following conditions" part, but I don't think "which algorithm happens to have the aliases referred to in this portion of the constraints" should be controlling here. It is more important that the constraints be contiguous.

Reply:
> I would also be OK with moving the other half into the SortIndexedProperties clause, I suppose, though that seems marginally worse.

Of the two suggestions, I dislike that one far less, so I've added a fixup to that effect. (Compromise!)

> [...] I don't think "which algorithm happens to have the aliases referred to in this portion of the constraints" should be controlling here. It is more important that the constraints be contiguous.

Understood, I just disagree. I'm not saying that locality-of-aliases should always win, only that it I think it gives the best outcome in this case.